test/lsp.py

```diff
@@ -898,18 +899,29 @@ def main(self) -> int:
 
         return min(max(self.test_counter.failed, self.assertion_counter.failed), 127)
 
-    def setup_lsp(self, lsp: JsonRpcProcess, expose_project_root=True):
+    def setup_lsp(
+        self,
+        lsp: JsonRpcProcess,
+        expose_project_root=True,
+        analyze_all_files_in_project=False,
+        project_root_suffix = None
+    ):
         """
         Prepares the solc LSP server by calling `initialize`,
         and `initialized` methods.
         """
+        rootUri = self.project_root_uri
+        if project_root_suffix is not None:
+            rootUri = rootUri + '/' + project_root_suffix
```

Comment:
Is this meant to be `/` on all platforms or should have a platform specific separator here? I'm guessing the latter - pathlib has `/` operator for that:
```suggestion
            rootUri = rootUri / project_root_suffix
```

Reply:
@cameel it's a string containing a URI, which is always using forward slash for path separation. So simply using '/' is fine. I'd need to extract the path part off `rootUri` first before I could make use of the `/` operator from PathLib in order to append a sub path.

Reply:
Call summary: not worth it (for now)

Reply:
To add some detail: it turned out that `rootUri` is a string, not `pathlib.Path` as I thought.